### PR TITLE
chore: support Kubernetes v1.25.7

### PR DESCRIPTION
**Pull request by jadarsie**
**Reason for Change**:

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.25.md#changelog-since-v1240

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/common/versions.go [hunk 1]
```diff
@@ -396,6 +397,7 @@ var AllKubernetesSupportedVersionsAzureStack = map[string]bool{
 	"1.23.15": true,
 	"1.24.7":  false,
 	"1.24.9":  true,
+	"1.25.7":  true,
 }
```
Comment (christinalau0) [inline pkg/api/common/versions.go:401]: Are we leaving 1.23.15 as supported for now? Plans to remove in the future?

Comment (jadarsie) [reply pkg/api/common/versions.go:401]: will do now

Comment (christinalau0) [reply pkg/api/common/versions.go:401]: Thanks, also need to change the defaults to 1.24 here then: https://github.com/Azure/aks-engine-azurestack/blob/master/pkg/api/common/const.go#L58

**Review by christinalau0 — APPROVED** [review]

**Pull request merged**